### PR TITLE
kubetest: re-merge aks deployer

### PR DESCRIPTION
**Pull request by alexeldeib**
...and don't break gce!

### CODE REVIEW

**Comment by alexeldeib** [comment]
/assign @feiskyer @mborsz 

(mborz -- hoping you can provide some eyes to make sure this doesn't affect gce again).

File: kubetest/e2e.go [hunk 1]
```diff
@@ -615,33 +620,52 @@ func kubemarkUp(dump string, o options, deploy deployer) error {
...
-		"gcloud", "compute", "addresses", "describe",
-		os.Getenv("MASTER_NAME")+"-ip",
-		"--project="+o.gcpProject,
-		"--region="+o.gcpZone[:len(o.gcpZone)-2],
-		"--format=value(address)"))
-	if err != nil {
-		return fmt.Errorf("failed to get masterIP: %v", err)
+	var masterIP, masterInternalIP []byte
+
+	if o.deployment == "bash" && o.provider == "gce" {
```
Comment (alexeldeib) [inline kubetest/e2e.go]: n.b.

File: kubetest/build.go [hunk 2]
```diff
@@ -50,7 +50,7 @@ func (b *buildStrategy) Set(value string) error {
 		}
 	}
 	switch value {
-	case "bazel", "e2e", "host-go", "quick", "release", "gce-windows-bazel", "none":
+	case "bazel", "e2e", "host-go", "quick", "release", "gce-windows-bazel", "none", "":
```
Comment (feiskyer) [inline kubetest/build.go]: does "none" same as ""?

File: kubetest/e2e.go (hunk 1 above)
Comment (feiskyer) [reply kubetest/e2e.go]: any specical reasons to check `o.deployment == "bash"`?

File: pwd.txt [hunk 3]
```diff
@@ -0,0 +1 @@
+Xwn95f75
```
Comment (feiskyer) [inline pwd.txt]: what is this?

File: kubetest/build.go (hunk 2 above)
Comment (alexeldeib) [reply kubetest/build.go]: yep, removed.

File: kubetest/e2e.go (hunk 1 above)
Comment (alexeldeib) [reply kubetest/e2e.go]: https://github.com/kubernetes/test-infra/pull/16452#discussion_r390901322

**Comment by alexeldeib** [comment]
/hold

since i'm breaking the kubemark bits out, we can simplify this a bit

**Comment by alexeldeib** [comment]
(leaving hold until I manually validate the cleanup)

**Comment by alexeldeib** [comment]
/hold cancel

seems okay, @feiskyer @mborsz could I ask you to take another look post-rebase?

File: kubetest/e2e.go [hunk 4]
```diff
@@ -587,11 +587,16 @@ func nodeTest(nodeArgs []string, testArgs, nodeTestArgs, project, zone string) e
...
-		return control.FinishRunning(exec.Command("./test/kubemark/stop-kubemark.sh"))
+		if err := control.FinishRunning(exec.Command("./test/kubemark/stop-kubemark.sh")); err != nil {
+			return fmt.Errorf("failed to stop kubemark cluster, err: %v", err)
+		}
+		return nil
 	}); err != nil {
 		return err
 	}
 
+	fmt.Println("finished tearing down kubemark")
```
Comment (mborsz) [inline kubetest/e2e.go]: I think that you should use log.Printf here instead of fmt.Println to be consistent with other print calls in this file.

**Comment by mborsz** [comment]
One tiny comment. LGTM otherwise

**Comment by alexeldeib** [comment]
think all review feedback is fixed, and re-fixed merge conflicts

File: kubetest/e2e.go (hunk 1 above)
Comment (feiskyer) [reply kubetest/e2e.go]: thanks

**Pull request merged**